electrum_nmc/electrum/names.py

```diff
@@ -131,6 +133,62 @@ def validate_value_length(value: bytes):
     value_length = len(value)
     if value_length > value_length_limit:
         raise BitcoinException('value length {} exceeds limit of {}'.format(value_length, value_length_limit))
+    
+def validate_onion_address(address: str) -> None: # The Address should not have .onion (An implementation to check & remove can be considered)
```

Comment:
I think we do want the input to have the `.onion` suffix.  Please verify that the suffix is present, and then remove it.